cdc/sink/codec/avro.go

```diff
@@ -152,9 +155,17 @@ func (a *AvroEventBatchEncoder) Size() int {
 }
 
 func (a *AvroEventBatchEncoder) avroEncode(table *model.TableName, tableVersion uint64, cols []*model.Column) (*avroEncodeResult, error) {
-	avroCodec, registryID, err := a.valueSchemaManager.Lookup(context.Background(), *table, tableVersion)
+	schemaGen := func() (string, error) {
+		schema, err := ColumnInfoToAvroSchema(table.Table, extractColumnInfos(cols))
+		if err != nil {
+			return "", errors.Annotate(err, "AvroEventBatchEncoder: generating schema failed")
+		}
+		return schema, nil
+	}
+
+	avroCodec, registryID, err := a.valueSchemaManager.GetCachedOrRegister(context.Background(), *table, tableVersion, schemaGen)
```

Comment:
We'd better pass the `ctx` from upper function, seems the interface of `EventBatchEncoder` should be udpated

Reply:
@liuzix